crates/sui-types/src/transaction.rs

```diff
@@ -2628,6 +2569,54 @@ pub type TrustedTransaction = TrustedEnvelope<SenderSignedData, EmptySignInfo>;
 pub type SignedTransaction = Envelope<SenderSignedData, AuthoritySignInfo>;
 pub type VerifiedSignedTransaction = VerifiedEnvelope<SenderSignedData, AuthoritySignInfo>;
 
+impl Transaction {
+    pub fn verify_signature(
+        &self,
+        current_epoch: EpochId,
+        verify_params: &VerifyParams,
+    ) -> SuiResult {
+        verify_sender_signed_data_message_signatures(self.data(), current_epoch, verify_params)
+    }
+
+    pub fn verify(
+        self,
+        current_epoch: EpochId,
+        verify_params: &VerifyParams,
+    ) -> SuiResult<VerifiedTransaction> {
+        self.verify_signature(current_epoch, verify_params)?;
+        Ok(VerifiedTransaction::new_from_verified(self))
+    }
+}
+
+impl SignedTransaction {
+    pub fn verify_signatures_authenticated(
```

Comment:
Probably need better naming to distinguish these two.
Maybe add `_into` to the consuming version

Reply:
Done, made it `try_into_verified` since its faillable